qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,270 @@
+            disp: Set to True to print convergence messages.
+        """
+
+        self.disp = disp
+        self.size_gen = size_gen
+        self.max_iter = maxiter
+        self.alpha = alpha
+        self.n_variables = n_variables
+        self.vector = self._initialization()
+        self.dead_iter = self.max_iter / 5
+
+        self.best_mae_global = 999999999999
+        self.truncation_length = int(size_gen * alpha)
+
+        # initialization of generation
+        self.generation = algorithm_globals.random.normal(
+            self.vector[0, :], self.vector[1, :], [self.size_gen, self.n_variables]
+        )
+
+        super().__init__(method="UMDA")
+
+    def _initialization(self):
+        vector = np.zeros((4, self.n_variables))
+
+        vector[0, :] = np.pi  # mu
+        vector[1, :] = 0.5  # std
+        vector[2, :] = 0  # min
+        vector[3, :] = np.pi * 2  # max
+
+        return vector
+
+    # build a generation of size SIZE_GEN from prob vector
+    def _new_generation(self):
+        """Build a new generation sampled from the vector of probabilities.
+        Updates the generation pandas dataframe
+        """
+
+        gen = algorithm_globals.random.normal(
+            self.vector[0, :], self.vector[1, :], [self.size_gen, self.n_variables]
+        )
+
+        self.generation = self.generation[: int(self.elite_factor * len(self.generation))]
+        self.generation = np.vstack((self.generation, gen))
+
+    # truncate the generation at alpha percent
+    def _truncation(self):
+        """Selection of the best individuals of the actual generation.
+        Updates the generation by selecting the best individuals.
+        """
+        best_indices = self.evaluations.argsort()[: self.truncation_length]
+        self.generation = self.generation[best_indices, :]
+        self.evaluations = np.take(self.evaluations, best_indices)
+
+    # check each individual of the generation
+    def _check_generation(self, objective_function):
+        """Check the cost of each individual in the cost function implemented by the user."""
+        self.evaluations = np.apply_along_axis(objective_function, 1, self.generation)
+
+    # update the probability vector
+    def _update_vector(self):
+        """From the best individuals update the vector of normal distributions in order to the next
+        generation can sample from it. Update the vector of normal distributions
+        """
+
+        for i in range(self.n_variables):
+            self.vector[0, i], self.vector[1, i] = norm.fit(self.generation[:, i])
+            if self.vector[1, i] < self.std_bound:
+                self.vector[1, i] = self.std_bound
+
+    def minimize(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> OptimizerResult:
+
+        self.history = []
```

Comment:
I commented on history etc as class variables. Here for example it will delete that history so if you had a prior instance you run that you wanted to compare its past history to a new instance that was running you could not as the prior instance'd history is this instances history too as they share a class variable.

Reply:
That is true. History has actually been initialized in the __init__ so I will delete the line 234.